.github/workflows/prebuild-deps.yml

```diff
@@ -0,0 +1,21 @@
+---
+## This workflow is used to prebuild and cache some dependencies for the build process.
+name: Prebuild dependencies
+on:
+  push:
+    branches:
+      - master
+      - build/optimize-memory # todo: remove this line before merging
+  schedule:
+    - cron: "30 4 * * 0"
+jobs:
+  deps-rocksdb:
+    name: Prebuild and cache librocksdb
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
```

Comment:
We must switch to AWS cache otherwise it's not going to work https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

Another option is bring back build on merging PRs to default branch (or scheduled run). But it make sense only if we switch completely to default runners.

Reply:
switched to s3